source/_integrations/valve.mqtt.markdown

```diff
@@ -0,0 +1,372 @@
+  default: 100
+qos:
+  description: The maximum QoS level to be used when receiving and publishing messages.
+  required: false
+  type: integer
+  default: 0
+reports_position:
+  description: "Set to `true` if the value reports the position or supports setting the position. Enabling the `reports_position` option will cause the position to be published instead of a payload defined by `payload_open`, `payload_close` or `payload_stop`. When receiving messages `state_topic`, will accept numeric payloads or a state message: `open`, `opening`, `closed`, or `closing`."
+  required: false
+  type: boolean
+  default: false
+retain:
+  description: Defines if published messages should have the retain flag set.
+  required: false
+  type: boolean
+  default: false
+state_closed:
+  description: The payload that represents the closed state. Is only allowed when `reports_position` is set to `False` (default).
+  required: false
+  type: string
+  default: closed
+state_closing:
+  description: The payload that represents the closing state.
+  required: false
+  type: string
+  default: closing
+state_open:
+  description: The payload that represents the open state. Is only allowed when `reports_position` is set to `False` (default).
+  required: false
+  type: string
+  default: open
+state_opening:
+  description: The payload that represents the opening state.
+  required: false
+  type: string
+  default: opening
+state_topic:
+  description: The MQTT topic subscribed to receive valve state messages. State topic accepts a state payload (`open`, `opening`, `closed`, or `closing`) or, if `reports_position` is supported, a numeric value representing the position. In a JSON format with variables `state` and `position` both values can received together.
+  required: false
+  type: string
+unique_id:
+  description: An ID that uniquely identifies this valve. If two valves have the same unique ID, Home Assistant will raise an exception.
+  required: false
+  type: string
+value_template:
+  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) that can be used to extract the payload for the `state_topic` topic. The rendered value should be a defined state payload or, if reporting a `position` is supported and `reports_position` is set to `true`, a numeric value is expected representing the position. See also `state_topic`."
+  required: false
+  type: template
+{% endconfiguration %}
+
+<div class="note">
+
+MQTT valve expects position values to be in range of 0 to 100, where 0 indicates closed position and 100 indicates fully open position.
+If `position_open` or `position_closed` are set to a different range (e.g. 40 to 140), when sending command to the device the range will be adjusted to the device range (position 0 will send a value of 40 to device) and when position payload is received from the device it will be adjusted back to the 0 to 100 range (device value of 40 will report valve position 0).
```

Comment:
```suggestion
If `position_open` or `position_closed` are set to a different range (for example, 40 to 140), when sending a command to the device, the range will be adjusted to the device range. For example, position 0 will send a value of 40 to device. When the device receives a position payload, it will be adjusted back to the 0 to 100 range. In our example, the device value of 40 will report valve position 0.
```